lib/elixir/lib/enum.ex

```diff
@@ -3903,8 +3903,11 @@ defmodule Enum do
       iex> Enum.zip([1, 2, 3, 4, 5], [:a, :b, :c])
       [{1, :a}, {2, :b}, {3, :c}]
 
+      iex> Enum.zip([], [:a, :b, :c])
+      []
+
   """
-  @spec zip(t, t) :: [{any, any}]
+  @spec zip(t, t) :: [{any, any}] | []
```

Comment:
`[t]` is like `list(t)`, not `nonempty_list(t)`, so this is unnecessary
```suggestion
  @spec zip(t, t) :: [{any, any}]
```

Reply:
Missed that, thanks!